File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        String t = s.toLowerCase();
        t = t.replaceAll(
          "[^a-zA-Z0-9]", "");
        int left = 0, right = t.length() - 1;
        while (left < right) {
            if (t.charAt(left) != t.charAt(right)) {
                return false; 
            }
            left++;
            right--;
        }
        
        return true;
        
    }
}

File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution(object):
    def isPalindrome(self, s):
        t = s.lower()
        t = ''.join(char for char in t if char.isalnum())
        
        left, right = 0, len(t) - 1
        while left < right:
            if t[left] != t[right]:
                return False  
            left += 1
            right -= 1
        
        return True
        